src/main/ruby/core/math.rb

```diff
@@ -15,6 +15,15 @@ module Math
 
   module_function
 
+  def hypot(a, b)
+    if Fixnum === a and Fixnum === b
```

Comment:
`&&` is conventional in `if`, `and` is another control-flow operator

Reply:
If we take such a rule, I think we should standardize it across the whole core and have RuboCop check it. Both our core and MRI's stdlib seem around 50% `if ... and`/`if ... &&` so there is no clear majority.
https://github.com/bbatsov/ruby-style-guide#no-and-or-or is not convincing me at all, it seems arbitrarily assuming people do not want to know about precedence, yet it's needed in Ruby for many others constructs like `{ ... }`/`do ... end`.
And a control-flow `and`/`or` is just heresy from Perl/PHP imho.